### ISSUE TITLE
Codegen error UDT declare with .twin and .bas

### ISSUE THREAD

**Issue by Kr00l** [WaynePhillipsEA/twinbasic#632]
**Describe the bug**
The below Sub Main will generate a compile error: codegen error.

**To Reproduce**
```
Module MainModule
  
  Private Type ISomething
  This As Long
  IID As t_IID
  End Type
  Private mIID As ISomething
  Private Declare Sub CLSIDFromString Lib "ole32" (ByVal psIID As Long, pIID As Any)
  
  Public Sub Main()
  
  CLSIDFromString StrPtr("{BB2DECE9-1C2F-4409-9141-23D81E70FA8B}"), mIID.IID
    
  End Sub
	
End Module
```

Put below in a legacy .bas file
```
Attribute VB_Name = "modCOMFactory"

Public Type t_IID
  D1 As Currency
  D2 As Currency
End Type
```

**Expected behavior**
to have no run-time error.

**Screenshots**
![image](https://user-images.githubusercontent.com/33756666/148023888-01ad78d9-9f89-44ed-8880-758bdbcc8dc5.png)

**Desktop (please complete the following information):**
 - OS: [e.g. Windows 10]
 - twinBASIC compiler version [e.g. 0.13.61]

**Additional context**
The problem can be circumvented by putting the t_IID into the .twin file as example below.
```
Module MainModule
  
  Private Type ISomething
  This As Long
  IID As t_IID
  End Type
  Private mIID As ISomething
  Private Declare Sub CLSIDFromString Lib "ole32" (ByVal psIID As Long, pIID As Any)
  
  Public Sub Main()
  
  CLSIDFromString StrPtr("{BB2DECE9-1C2F-4409-9141-23D81E70FA8B}"), mIID.IID
    
  End Sub
	
End Module

Module AnotherModule
  Public Type t_IID
    D1 As Currency
    D2 As Currency
  End Type
End Module
```

**Comment by WaynePhillipsEA** [comment]
Thanks for reporting.   We can repro within a single twin file, by moving AnotherModule above the MainModule in your example.    This is due to #190.

**Comment by WaynePhillipsEA** [comment]
With #190 now fixed, this is also now fixed in v0.14.10.  Thanks for reporting!
